### PR TITLE
Provide context to initial build of subgraph, surface API

### PR DESCRIPTION
**Pull request by MingweiSamuel**

### CODE REVIEW

File: hydroflow/src/scheduled/graph.rs [hunk 1]
```diff
@@ -425,6 +424,16 @@ impl Hydroflow {
...
+        self.context.subgraph_id = sg_id;
+        &mut self.context
+    }
+
+    pub(crate) fn next_subgraph_id(&self) -> SubgraphId {
+        SubgraphId(self.subgraphs.len())
+    }
+
     pub fn add_dependencies(&mut self, sg_id: SubgraphId, deps: FlowGraph) {
         self.subgraphs[sg_id.0].dependencies.append(deps);
```
Comment (MingweiSamuel) [inline hydroflow/src/scheduled/graph.rs:438]: changes

File: hydroflow/src/builder/hydroflow_builder.rs [hunk 2]
```diff
@@ -109,6 +112,8 @@ impl HydroflowBuilder {
                 pivot.run();
             },
         );
+        assert_eq!(sg_id, sg_id_actual);
```
Comment (MingweiSamuel) [inline hydroflow/src/builder/hydroflow_builder.rs:115]: Changes here

Comment (justinj) [reply hydroflow/src/builder/hydroflow_builder.rs:115]: I can't think of a better way to do this with the current architecture but it seems unfortunate that we have to do this concurrently and then compare them :(

**Review by justinj — APPROVED** [review]

File: hydroflow/src/builder/hydroflow_builder.rs (hunk 2 above)
Comment (MingweiSamuel) [reply hydroflow/src/builder/hydroflow_builder.rs:115]: I agree

**Pull request merged**